src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -236,12 +237,18 @@ def slice_info_for_literal(self, length, prefix="") -> RawSliceInfo:
             unique_alternate_id, alternate_code, inside_block=self.inside_block
         )
 
-    def update_inside_set_or_macro_or_block(
-        self, block_type: str, trimmed_parts: List[str]
-    ) -> None:
-        """Based on block tag, update whether we're in a set/macro section."""
+    def update_inside_set_call_macro_or_block(
```

Comment:
I wonder if there's a more generic name we could use here? Rather than adding to the function name each time we find a new jinja block we want to include. Maybe `update_inside_jinja_block`?

Reply:
Good point. I'll leave it like this for now. At some point, I think we should refactor this code a bit, so each kind of block is handled by its own Python class, like the parser segments. It would be a big reorganization, but would make the code easier to work on. I keep thinking we must've already handled everything in Jinja, but every few months, I learn that's not the case...